### PR TITLE
1476817: Set network.ipv4_address properly, when DNS misconfigured.

### PR DESCRIPTION
**Pull request by jirihnidek**
* Bug fix: https://bugzilla.redhat.com/show_bug.cgi?id=1476817
* When there is no record for hostname in DNS or /etc/hosts, then
  set network.ipv4_address from list of addresses of all
  interfaces in the system (loopback is skipped).
* Do similar for network.ip6_address, but use only interfaces
  with global addresses.

### CODE REVIEW

File: src/rhsmlib/facts/hwprobe.py [hunk 1]
```diff
@@ -565,6 +565,40 @@ def get_ls_cpu_info(self):
...
+        """
+        When DNS record is not configured properly for the system, then try to
+        get list of all IPv4 addresses from all devices. Return 127.0.0.1 only
+        in situation when there is only loopback device.
+        :return: list of IPv4 addresses
+        """
+        addr_list = []
+        interface_info = ethtool.get_interfaces_info(ethtool.get_devices())
+        for info in interface_info:
+            for addr in info.get_ipv4_addresses():
```
Comment (awood) [inline src/rhsmlib/facts/hwprobe.py]: Could we just do

```python
addr_list = [addr for addr in info.get_ipv4_addresses() if addr.address != '127.0.0.1']
```

Comment (jirihnidek) [reply src/rhsmlib/facts/hwprobe.py]: I prefer list comprehension too, but only for simple situation. There are two for loops. I can rewrite it using list comprehension:

```python
addr_list = [addr.address for info in interface_info for addr in info.get_ipv4_addresses() if addr.address != '127.0.0.1']
```

but I consider original code more clear and readable ;-).

Comment (awood) [reply src/rhsmlib/facts/hwprobe.py]: Okay.  Looks good to me otherwise.  Those Jenkins failures are false positives I believe.

**Comment by awood** [comment]
bot, retest this please

**Pull request merged**